website/eslint.config.js

```diff
@@ -0,0 +1,146 @@
+    '@typescript-eslint/consistent-type-definitions': 'off',
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/consistent-indexed-object-style': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/prefer-reduce-type-parameter': 'off',
+    '@typescript-eslint/no-empty-function': 'off',
+    '@typescript-eslint/triple-slash-reference': 'off',
+};
+
+const namingConvention = {
+    '@typescript-eslint/naming-convention': [
+        'error',
+        {
+            selector: 'default',
+            format: ['camelCase'],
+            leadingUnderscore: 'allow',
+            trailingUnderscore: 'allow',
+        },
+        {
+            selector: 'function',
+            format: ['camelCase', 'PascalCase'],
+        },
+        {
+            selector: 'variable',
+            format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+            leadingUnderscore: 'allow',
+            trailingUnderscore: 'allow',
+        },
+        {
+            selector: 'enumMember',
+            format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+        },
+        {
+            selector: 'import',
+            format: null,
+        },
+        {
+            selector: 'typeLike',
+            format: ['PascalCase'],
+        },
+    ],
+};
+
+const restrictTemplateExpressions = {
+    '@typescript-eslint/restrict-template-expressions': [
+        'error',
+        {
+            allowNumber: true,
+            allow: [{ name: ['unknown', 'Error', 'URLSearchParams', 'URL'], from: 'lib' }],
+        },
+    ],
+};
+
+const disableFromReact = {
+    'react/no-unescaped-entities': 'off',
+    'react/display-name': 'off',
+    'react/react-in-jsx-scope': 'off',
+};
+
+export default tseslint.config(
+    { ignores: ['dist'], files: ['**/*.ts', '**/*.tsx'] },
+    {
+        files: ['**/*.ts', '**/*.tsx'],
+        extends: [
+            eslint.configs.recommended,
+            ...tseslint.configs.strictTypeChecked,
+            ...tseslint.configs.stylisticTypeChecked,
+        ],
+        languageOptions: {
+            parser: parser,
+            parserOptions: {
+                projectService: true,
+                tsconfigRootDir: import.meta.dirname,
+                ecmaFeatures: {
+                    jsx: true,
+                },
+            },
+        },
+        plugins: {
+            react,
+            'import': importPlugin,
+            'react-hooks': reactHooks,
+        },
+        rules: {
+            ...react.configs.flat.recommended.rules,
+            ...importRules,
+            ...namingConvention,
+            ...restrictTemplateExpressions,
+            ...disableFromTypescriptEsLint,
+            ...disableFromReact,
+            ...enableFromEslint,
+        },
+        settings: {
+            react: {
+                version: 'detect',
+            },
+        },
+    },
+    {
+        files: ['**/*.astro'],
```

Comment:
Is there a reason why we don't use the `importPlugin` here?

Reply:
I disabled it, since the importPlugin gives us the false positives for the transitive imports, like in astro-auth. I can also disable the specific rule?
And it also provides us with some parsing error like:
```
Error while parsing .../covSpectrum/dashboards/website/node_modules/react-toastify/dist/react-toastify.esm.mjs
Line 4, column 0: Unknown token at 15459, expected: "}", actual: ""
`parseForESLint` from parser `context.languageOptions.parser` is invalid and will just be ignored
```